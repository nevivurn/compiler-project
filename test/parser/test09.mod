//
// test09
//
// parser test:
// - control flow statements
//
module test09;

var a,b,c: integer;

begin
  a:= 0; b := 1; c := 2;

  if (a > b) then
    c := a - b
  end;

  if (b > c) then
    a := b - c
  else
    a := c - b
  end;

  while (a > 0) do
    a := a - 1
  end
end test09.
