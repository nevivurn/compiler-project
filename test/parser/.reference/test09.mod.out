parsing 'test09.mod'...
successfully parsed.
  AST:
    CAstScope: 'test09'
      symbol table:
        [[
          [ *DIM(<ptr(8) to <NULL>>,<integer>) --> <integer>           ]
          [ *DOFS(<ptr(8) to <NULL>>) --> <integer>           ]
          [ *ReadInt() --> <integer>           ]
          [ *ReadLong() --> <longint>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<integer>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteLong(<longint>) --> <NULL>           ]
          [ *WriteStr(<ptr(8) to array of <char>>>) --> <NULL>           ]
          [ @a        <integer>           a           ]
          [ @b        <integer>           b           ]
          [ @c        <integer>           c           ]
          [ main     <NULL>           ]
        ]]
      statement list:
        := <integer>
          [ @a        <integer>           a ] <integer>
          0 <integer>
        := <integer>
          [ @b        <integer>           b ] <integer>
          1 <integer>
        := <integer>
          [ @c        <integer>           c ] <integer>
          2 <integer>
        if cond
          > <boolean>
            [ @a        <integer>           a ] <integer>
            [ @b        <integer>           b ] <integer>
        if-body
          := <integer>
            [ @c        <integer>           c ] <integer>
            sub <integer>
              [ @a        <integer>           a ] <integer>
              [ @b        <integer>           b ] <integer>
        else-body
          empty.
        if cond
          > <boolean>
            [ @b        <integer>           b ] <integer>
            [ @c        <integer>           c ] <integer>
        if-body
          := <integer>
            [ @a        <integer>           a ] <integer>
            sub <integer>
              [ @b        <integer>           b ] <integer>
              [ @c        <integer>           c ] <integer>
        else-body
          := <integer>
            [ @a        <integer>           a ] <integer>
            sub <integer>
              [ @c        <integer>           c ] <integer>
              [ @b        <integer>           b ] <integer>
        while cond
          > <boolean>
            [ @a        <integer>           a ] <integer>
            0 <integer>
        while-body
          := <integer>
            [ @a        <integer>           a ] <integer>
            sub <integer>
              [ @a        <integer>           a ] <integer>
              1 <integer>
      nested scopes:
        empty.
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -otest09.mod.ast.pdf test09.mod.ast.dot


Done.
